### DISCUSSION TITLE
Cloud Edition and Community Edition difference

### DISCUSSION THREAD

**Discussion by sgfisme** [frePPLe/frepple#444] [Q&A]
Hi, I tried to install community edition on local docker environment. And use the same data for testing. But the plan result seemed like different on cloud environment. 

Just wondering to know what's different between cloud and community edition. E.g., solver? planning type? 
Or any hardware suggestion on container. E.g., CPU/RAM? size of volume?

**Comment by jdetaeye** [comment]
>> Just wondering to know what's different between cloud and community edition. E.g., solver? planning type?

See https://frepple.com/editions/ for the high level comparison.
The Enterprise & Cloud Editions are based on the same core solver as the Community Edition, and add extra functionality on top of it.  
You should get the same plan as with the Community Edition, except in the following situations:
- you have approved production orders: the community edition cannot reschedule them as in the enterprise edition
- you have setup matrices:  the Enterprise&Cloud editions have extra capabilities to optimize changeovers on resources
- you have infeasibilities in your work-in-progress information:   the Enterprise&Cloud editions have extra functionality to handle those

You can expect the differences to grow in the future: the user community (which includes YOU and EVERYONE READING THIS) doesn't contribute enough to the project. So, why would the frepple team time & energy in the community edition without a real win-win situation?   

>> Or any hardware suggestion on container. E.g., CPU/RAM? size of volume?

That's a completely different topic :-)  It definitely doesn't explain any difference in the plan results.  

We should definitely improve the documentation of the CPU&RAM requirements.
In general, frepple isn't particularly resource hungry. Even a modest laptop computer can support frepple instances of a decent size.

**Comment by nameduser0** [comment]
## TL;DR: almost all features advertised on [frepple's website](https://frepple.com/features/) are NOT available in the Community Edition.

So based on the edition [comparison](https://frepple.com/editions/) (BTW you really need to use a HTML table than try to align divs, but I think that's because you want to obfuscate the differences between Community and Enterprise editions as much as possible, just like Odoo does), these features are **missing** from the Community Edition:

- Demand forecasting module: you can't see the forecast
- Inventory planning module: you can't see your inventory situation
- Interactive Gantt chart: you can't see the created plan
- Order due date quoting: you can't find out if a new order delivery date is achievable
- Resource (machine) setup optimisation

# Menu comparison: Community Edition on the left, Enterprise on the right
## Sales - Forecast and Quotes missing
![Sales-CE](https://user-images.githubusercontent.com/64409021/158842509-5dbf9d99-3cf5-414e-b627-a4ae1028b1a3.png) ![Sales-EN](https://user-images.githubusercontent.com/64409021/158842529-ae09cefc-ef24-4ba9-b0df-87056bdc5ee7.png)

## Inventory - Inventory options missing
![Inventory-CE](https://user-images.githubusercontent.com/64409021/158842552-346b0822-393e-406a-b448-d2e7ef55c985.png) ![Inventory-EN](https://user-images.githubusercontent.com/64409021/158842572-62309c1b-5377-45f2-9eb1-e0b6c01c498c.png)

## Capacity - Resource options missing
![Capacity-CE](https://user-images.githubusercontent.com/64409021/158842595-114b3bff-b457-410c-93aa-935ee2bfee19.png) ![Capacity-EN](https://user-images.githubusercontent.com/64409021/158842613-81fc12ac-7163-40d6-b2a4-24b0052dee41.png)

## Purchasing
![Purchasing-CE](https://user-images.githubusercontent.com/64409021/158842663-d2ee4907-44a7-4ec1-9e75-d041cf0a6a81.png) ![Purchasing-EN](https://user-images.githubusercontent.com/64409021/158842682-98881e63-2d47-48ee-a144-082fcc5d391e.png)

## Manufacturing - no Plan Editor
![Manufacturing-CE](https://user-images.githubusercontent.com/64409021/158842711-cf0b627e-f8d0-4887-a561-4c795e1142d1.png) ![Manufacturing-EN](https://user-images.githubusercontent.com/64409021/158842730-3968aa5e-6d17-4694-b434-08627f7655a0.png)

## Admin
![Admin-CE](https://user-images.githubusercontent.com/64409021/158842754-f78ba17b-ac17-49f5-abc5-c62c85741a54.png) ![Admin-EN](https://user-images.githubusercontent.com/64409021/158842770-c50fda5e-235b-4323-b687-d34eb834e0e6.png)

## Plan Execution - options missing
![Execute-CE](https://user-images.githubusercontent.com/64409021/158842802-b5585a86-6dac-493f-8a45-3ae221678788.png) ![Execute-EN](https://user-images.githubusercontent.com/64409021/158842821-f359f670-4931-4198-a928-6571fb6d3067.png)

> the user community (which includes YOU and EVERYONE READING THIS) doesn't contribute enough to the project. So, why would the frepple team time & energy in the community edition without a real win-win situation?

As a counter-argument: why would the open-source community spend time improving your product when you withhold almost all of it's useful features? Do you expect us to implement the features you have already developed only to have them NOT integrated and rejected in the Community Edition because it would damage your profits? We would rather spend time implementing them in a true open-source project.

Or are you promising to release more features if we provide integrations? I notice you expect the community to provide language translations for FREE.
There has recently been an ERPNext integration released. Which features are you putting into the Community Edition for this?

**Reply by jdetaeye** [reply]
>the user community (which includes YOU and EVERYONE READING THIS) doesn't contribute enough to the project. So, why would the frepple team time & energy in the community edition without a real win-win situation?
>
>> As a counter-argument: why would the open-source community spend time improving your product when you withhold almost all of it's useful features? Do you expect us to implement the features you have already developed only to have them NOT integrated and rejected in the Community Edition because it would damage your profits? We would rather spend time implementing them in a true open-source project.

Open source comes in many flavors, colors and sizes:  https://en.wikipedia.org/wiki/Business_models_for_open-source_software.   
We have chosen the open-core business model, and are happy with that choice.

>> Or are you promising to release more features if we provide integrations? I notice you expect the community to provide language translations for free.

A more active community will definitely also make it more attractive for us to move the boundary. 

>> There has recently been an ERPNext integration released. Which features are you putting into the Community Edition for this?

That integration is not managed or maintained by the frepple team.  It's 100% open source on https://github.com/msf4-0/ERPNext-Frepple-Integration 
We'll contribute to it and promote it as third party component in the frepple documentation.  And if the integration requires some changes in frepple, we accept pull requests.

**Reply by nameduser0** [reply]
> That integration is not managed or maintained by the frepple team. It's 100% open source on https://github.com/msf4-0/ERPNext-Frepple-Integration
> We'll contribute to it and promote it as third party component in the frepple documentation. And if the integration requires some changes in frepple, we accept pull requests.

So in other words: you will accept contributions but won't give anything back except to expand your own product offering? He doesn't get anything from your promotion. So why do you bother making statements like:

> doesn't contribute enough to the project. So, why would the frepple team time & energy in the community edition without a real win-win situation?

What is your "win-win situation" exactly? How do I win?
